ejb3/src/main/java/org/jboss/as/ejb3/component/EJBComponentDescription.java

```diff
@@ -332,8 +332,8 @@ public void configure(final DeploymentPhaseContext context, final ComponentDescr
 
                 if (description.isTimerServiceRequired()) {
 
-                    if (!ejbSetupActions.isEmpty()) {
-                        configuration.addTimeoutViewInterceptor(AdditionalSetupInterceptor.factory(ejbSetupActions), InterceptorOrder.View.EE_SETUP);
+                    for (SetupAction setupAction : ejbSetupActions) {
+                        configuration.addTimeoutViewInterceptor(new ImmediateInterceptorFactory(new AdditionalSetupInterceptor(setupAction)), setupAction.priority() == 0? InterceptorOrder.View.EE_SETUP : setupAction.priority());
```

Comment:
@stuartwdouglas I was afraid of having future consequences if I changed the priority value, because I was unsure what were the reasons, if any, for having EE_SETUP priority set to its current value. If you think it is okay, I'll do as you're suggesting. Thanks!